examples/yaml-to-anydata/yaml_to_anydata.bal

```diff
@@ -0,0 +1,53 @@
+import ballerina/data.yaml;
+import ballerina/io;
+
+type ServerConfig record {|
+    string host;
+    int port;
+    string protocol;
+|};
+
+
+final string yamlString = string
+        `
+        host: "localhost"
+        port: 8080
+        protocol: "http"`;
+
+public function main() returns error? {
+    // Parse the YAML string to a record type.
+    ServerConfig serverConfig1 = check yaml:parseString(yamlString);
+    io:println(serverConfig1);
+
+    byte[] yamlByteArr = yamlString.toBytes();
+    // Parse the YAML byte array to a record type.
+    ServerConfig serverConfig2 = check yaml:parseBytes(yamlByteArr);
+    io:println(serverConfig2);
+
+    stream<byte[], error?> byteBlockStream = new (new StreamImplementor(yamlString));
+    // Parse the YAML byte block stream to a record type.
+    ServerConfig serverConfig3 = check yaml:parseStream(byteBlockStream);
+    io:println(serverConfig3);
+}
+
+// Defines a class called `StreamImplementor`, which implements the `next()` method.
```

Comment:
What I meant was to say something along the lines of Define a class that is a stream implementor, ...